tests/tests_unit/test_config_set.py

```diff
@@ -0,0 +1,73 @@
+from unittest.mock import patch
+
+import pytest
+from packaging.requirements import Requirement
+
+from _nebari.config_set import ConfigSetMetadata, read_config_set
+
+test_version = "2024.12.2"
+
+
+def test_valid_version_requirement():
+    metadata = ConfigSetMetadata(
+        name="test-config", nebari_version=">=2024.12.0,<2025.0.0"
+    )
+    assert metadata.nebari_version.specifier.contains(test_version)
+
+
+def test_invalid_version_requirement():
```

Comment:
This could be parameterized too to validate multiple invalid types

Reply:
done